### PR TITLE
Data download and setting environment

### PR DESCRIPTION
**Pull request by mjw-18**

### CODE REVIEW

**Comment by mjw-18** [comment]
@gwaybio Hey Greg! This pull request is now ready for review.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 .ipynb_checkpoints
+data/
```
Comment (gwaybio) [inline .gitignore]:
```suggestion
0.data-download/data/*

```

To be more specific about which data folder. We might want to include other data in other folders in the future.

Also, I'm adding an extra new line character to the end of the file. This removes the red warning label at the end of line two.

p.s. - See [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) for instructions on how to accept "suggestions" into your  PR.

File: 0.data-download/data_downloader.ipynb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (gwaybio) [inline 0.data-download/data_downloader.ipynb:1]: Jupyter notebook files (`.ipynb`) are [JSON formatted](https://ipython.org/ipython-doc/3/notebook/nbformat.html), which makes them difficult to review.

For this reason, we always include two copies of all notebooks:

1. the `.ipynb` version you've included here
2. an `nbconverted` `.py` version you need to add

To create the `.py` version, run the following command:

```bash
# Activate your conda environment
conda activate __NAME__

# Navigate to the proper folder
cd 0.data-download

# Run nbconvert to create a .py file
jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/nbconverted *.ipynb
```

Please add, commit, push this nbconverted file to this PR, and I will review this file.

File: LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+BSD 3-Clause License
+
+Copyright (c) 2022, Way Laboratory All rights reserved.
```
Comment (gwaybio) [inline LICENSE.md:3]: This is ok for now - thanks!

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+- conda-forge::ipython	=	8.4.0
+- conda-forge::ipython_genutils	=	0.2.0
+- conda-forge::jedi	=	0.18.1
+- conda-forge::jinja2	=	3.1.2
+- conda-forge::json5	=	0.9.5
+- conda-forge::jsonschema	=	4.6.0
+- conda-forge::jupyter_client	=	7.3.4
+- conda-forge::jupyter_core	=	4.10.0
+- conda-forge::jupyter_server	=	1.17.1
+- conda-forge::jupyterlab	=	3.4.3
```
Comment (gwaybio) [inline environment.yml:5]: Keep this package

File: environment.yml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+- conda-forge::psutil	=	5.9.1
+- conda-forge::ptyprocess	=	0.7.0
+- conda-forge::pure_eval	=	0.2.2
+- conda-forge::pycparser	=	2.21
+- conda-forge::pygments	=	2.12.0
+- conda-forge::pyopenssl	=	22.0.0
+- conda-forge::pyparsing	=	3.0.9
+- conda-forge::pyrsistent	=	0.18.1
+- conda-forge::pysocks	=	1.7.1
+- conda-forge::python	=	3.9.13
```
Comment (gwaybio) [inline environment.yml:6]: keep this line

File: environment.yml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- conda-forge::soupsieve	=	2.3.1
+- conda-forge::sqlite	=	3.38.5
+- conda-forge::stack_data	=	0.2.0
+- conda-forge::terminado	=	0.15.0
+- conda-forge::tinycss2	=	1.1.1
+- conda-forge::tk	=	8.6.12
+- conda-forge::tornado	=	6.1
+- conda-forge::traitlets	=	5.2.2.post1
+- conda-forge::tzdata	=	2022a
+- conda-forge::urllib3	=	1.26.9
```
Comment (gwaybio) [inline environment.yml:7]: keep this line

File: environment.yml [hunk 7]
```diff
@@ -0,0 +1,120 @@
+name: gene_dependency_representations
```
Comment (gwaybio) [inline environment.yml:1]: It is poor practice to add all packages to the conda environment - I have found that doing so is far too rigid, making "solving the environment" far to restricted, and is likely to fail in a short amount of time and/or in different operating systems. I have been guilty of this in the past!

Please delete all `dependencies` except the three I highlighted. Only add a critical dependency when you need to use a new package.

**Review by gwaybio — CHANGES_REQUESTED** [review]
Thanks for your PR @mjw-18 ! Nice work!

Let's discuss the several comments I made.

I'll review the data downloader notebook after you provide the nbconverted file (see comment below)

**Comment by mjw-18** [comment]
@gwaybio I believe I have performed the requested updates correctly! let me know if these check out!

**Comment by gwaybio** [comment]
i see the `.py` file, but I don't see the updates in the `environment.yml`. I'll take a look at the .py file now!

File: 0.data-download/scripts/nbconverted/data_downloader.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+# coding: utf-8
+
+# ## Download Dependency Data
+# 
+# Source: [Cancer Dependency Map resource](https://depmap.org/portal/download/).
+# 
+# - `CRISPR_gene_dependency.csv`: DESCRIBE HERE
+# - `sample_info.csv`: DESCRIBE HERE
+
+# In[2]:
```
Comment (gwaybio) [inline 0.data-download/scripts/nbconverted/data_downloader.py]: Looks like you will have to _reset kernel and run all_ again. This should be executed first.

File: environment.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+channels:
+- conda-forge
+dependencies:
+- conda-forge::jupyter_client	=	7.3.4
+- conda-forge::jupyter_core	=	4.10.0
+- conda-forge::jupyter_server	=	1.17.1
+- conda-forge::jupyterlab	=	3.4.3
```
Comment (gwaybio) [inline environment.yml]:
```suggestion
- conda-forge::jupyterlab=3.4.3
```

File: environment.yml [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+- conda-forge::jupyterlab_pygments	=	0.2.2
+- conda-forge::jupyterlab_server	=	2.14.0
+- conda-forge::pyparsing	=	3.0.9
+- conda-forge::pyrsistent	=	0.18.1
+- conda-forge::pysocks	=	1.7.1
+- conda-forge::python	=	3.9.13
```
Comment (gwaybio) [inline environment.yml]:
```suggestion
- conda-forge::python=3.9.13
```

File: environment.yml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+- conda-forge::tornado	=	6.1
+- conda-forge::traitlets	=	5.2.2.post1
+- conda-forge::tzdata	=	2022a
+- conda-forge::urllib3	=	1.26.9
```
Comment (gwaybio) [inline environment.yml]:
```suggestion
- conda-forge::urllib3=1.26.9

```

**Review by gwaybio — CHANGES_REQUESTED** [review]
Make sure your notebook starts with the first cell executed first (right now it is `2`), and trim your conda env file a bit more

**Review by gwaybio — APPROVED** [review]
Wonderful. Nice work!

**Pull request merged**